src/components/vm/consoles/vncBody.jsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright 2024 Fsas Technologies Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+    FormGroup, FormHelperText, HelperText, HelperTextItem,
+    InputGroup, TextInput, Button, Checkbox
+} from "@patternfly/react-core";
+import { EyeIcon, EyeSlashIcon } from "@patternfly/react-icons";
+
+import cockpit from 'cockpit';
+
+const _ = cockpit.gettext;
+
+export const VncRow = ({ idPrefix, onValueChanged, dialogValues, validationErrors }) => {
+    const [showPassword, setShowPassword] = useState(false);
+
+    return (
+        <>
+            <FormGroup
+                fieldId={`${idPrefix}-portmode`}
+                label={_("Listening")} isInline hasNoPaddingTop isStack>
+                <Checkbox
+                    id={`${idPrefix}-autoport`}
+                    label={_("Use a static port")}
+                    isChecked={dialogValues.vncCustomPort}
+                    onChange={(_ev, checked) => onValueChanged('vncCustomPort', checked)} />
+                { dialogValues.vncCustomPort &&
+                    <TextInput
+                        id={`${idPrefix}-port`}
+                        value={dialogValues.vncPort}
+                        type="text"
+                        validated={validationErrors.vncPort ? "error" : null}
+                        onChange={(event) => onValueChanged('vncPort', event.target.value)} />
+                }
+                { dialogValues.vncCustomPort && validationErrors.vncPort &&
+                    <FormHelperText>
+                        <HelperText>
+                            <HelperTextItem variant='error'>{validationErrors.vncPort}</HelperTextItem>
+                        </HelperText>
+                    </FormHelperText>
+                }
+            </FormGroup>
+            <FormGroup fieldId={`${idPrefix}-password`} label={_("Password")}>
```

Comment:
Does it actually make sense to offer a password for the "localhost only" use case? My gut feeling is "no", as that doesn't offer much protection in that use case. But @Shotaro-Kawaguchi should have the last word on that.

Reply:
This was discussed, and apparently there is an use case for password. So let's keep this in.

Reply:
Sure, in a multi-user situation.

Also, the "normal" way to set up remote access to VNC ports is via qemu.conf, I think. You would set a global listening address for all VMs ("hypervisor default" as virt-manager calls it), and also take care of encryption there. You can even set a global VNC password in qemu.conf.

Thus, even if Cockpit itself does not support changing the VNC listening address of a individual machine, we might not operate in a localhost-only setup.